server/listeners/eventHandlers/prisonEvents/datesChangedEventHandler.ts

```diff
@@ -18,27 +18,38 @@ export default class DatesChangedEventHandler {
 
     const nomisId =
       offenderIdDisplay ||
-      (await this.prisonerService.searchPrisonersByBookingIds([bookingId])).map(o => o.prisonerNumber).pop()
+      (!!bookingId &&
```

Comment:
could you push this check into the client?

Reply:
This is event handler, so I added it here.